Google.Api.Gax.Rest/ClientBuilderBase.cs

```diff
@@ -241,6 +242,43 @@ protected async virtual Task<IConfigurableHttpClientInitializer> GetHttpClientIn
         /// </summary>
         public abstract Task<TClient> BuildAsync(CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Populates properties based on those set via dependency injection.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Credentials are only requested from dependency injection if they are not already set
+        /// via any of <see cref="CredentialsPath"/>,
+        /// <see cref="JsonCredentials"/>, <see cref="Credential"/> or <see cref="GoogleCredential"/>.
+        /// </para>
+        /// <para>
+        /// If credentials are requested, they are tried in the following order:
+        /// </para>
+        /// <list type="bullet">
+        /// <item>ICredential</item>
+        /// <item>GoogleCredential</item>
+        /// </list>
+        /// </remarks>
+        /// <param name="provider">The service provider to request dependencies from.</param>
+        protected virtual void Configure(IServiceProvider provider)
```

Comment:
We need to remember to handwrite the extension methods for BigQuery and Storage clients.